### PR TITLE
Replace isDevEnvironment variable

### LINKED ISSUE #44: Replace isDevEnvironment boolean
## Description

Follow-up of https://github.com/cowprotocol/cow-sdk/pull/42#pullrequestreview-1026191339

### PR DESCRIPTION
**Pull request by matextrem**
This PR closes #44
#46 

Updated isDevEnvironment boolean variable to a better descriptive one.

### CODE REVIEW

File: src/api/cow/index.ts [hunk 1]
```diff
@@ -103,19 +103,19 @@ export class CowApi {
   }
 
   get API_BASE_URL() {
-    return getGnosisProtocolUrl(this.context.isDevEnvironment)
+    return getGnosisProtocolUrl(this.context.env === 'staging')
```
Comment (alfetopito) [inline src/api/cow/index.ts:106]: This function signature looks weird now.

A refactor suggestion is to pass the `env` as parameter, and on `getGnosisProtocolUrl`, use a `switch`.
When all types are covered you don't need to add a `default` clause and TS will be happy.

Something like:

```ts
switch(env) {
 case 'staging': return 'staging url....'
 case 'prod': return 'prod url..'
}

**Review by alfetopito — COMMENTED** [review]
Nitpick comments/suggestions

File: src/utils/context.ts [hunk 2]
```diff
@@ -102,8 +104,8 @@ export class Context implements Partial<CowContext> {
     return this.#context.appDataHash ?? DefaultCowContext.appDataHash
   }
 
-  get isDevEnvironment(): boolean {
-    return this.#context.isDevEnvironment ?? DefaultCowContext.isDevEnvironment
+  get env(): Env {
+    return this.#context.env ?? DefaultCowContext.env
```
Comment (W3stside) [inline src/utils/context.ts:108]: why the coalescing? this means `' '` (empty string) is ok, do we want that? maybe also why the class is forcing you to type cast as `Env` above

would suggest `this.#context.env || DefaultCowContext.env`

File: src/api/cow/index.ts (hunk 1 above)
Comment (ramirotw) [reply src/api/cow/index.ts:106]: `getGnosisProtocolUrl` -> `getCowProtocolUrl` ? does it make sense to do it in this PR?

File: src/utils/context.ts [hunk 3]
```diff
@@ -14,14 +16,14 @@ export interface Ipfs {
...
-  isDevEnvironment?: boolean
+  env?: Env
   signer?: Signer
   ipfs?: Ipfs
 }
 
 export const DefaultCowContext = {
   appDataHash: DEFAULT_APP_DATA_HASH,
-  isDevEnvironment: false,
+  env: 'staging' as Env,
```
[the remark this answers is not in the document]
Comment (matextrem) [reply src/utils/context.ts:26]: exactly

**Review by ramirotw — APPROVED** [review]

File: src/utils/context.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ramirotw) [reply src/utils/context.ts:26]: If we want to close #46 here shouldn't the default be `prod`?

[the remark this answers is not in the document]
Comment (alfetopito) [reply src/utils/context.ts:26]: So, are you addressing #46 here or in another PR?
Also, there still casting that I don't understand why is needed

**Review by alfetopito — CHANGES_REQUESTED** [review]
Oops, there's a unwanted file sneaking in
Screen Shot 2022-07-18 at 09 59 54

File: src/utils/context.ts [hunk 4]
```diff
@@ -102,8 +105,8 @@ export class Context implements Partial<CowContext> {
     return this.#context.appDataHash ?? DefaultCowContext.appDataHash
   }
 
-  get isDevEnvironment(): boolean {
-    return this.#context.isDevEnvironment ?? DefaultCowContext.isDevEnvironment
+  get env(): Env | undefined {
```
Comment (ramirotw) [inline src/utils/context.ts:108]: Could this be undefined?

**Review by alfetopito — APPROVED** [review]
👏

**Pull request merged**